Sources/BraveWallet/Crypto/Transaction Confirmations/PendingTransactionView.swift

```diff
@@ -0,0 +1,527 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import BraveCore
+import BigNumber
+import Strings
+import DesignSystem
+
+struct PendingTransactionView: View {
+  @ObservedObject var confirmationStore: TransactionConfirmationStore
+  let networkStore: NetworkStore
+  let keyringStore: KeyringStore
+  @Binding var isShowingGas: Bool
+  @Binding var isShowingAdvancedSettings: Bool
+  
+  var onClose: () -> Void
+  
+  @Environment(\.sizeCategory) private var sizeCategory
+  @Environment(\.openWalletURLAction) private var openWalletURL
+  
+  /// Blockie size for ERC 20 Approve transactions
+  @ScaledMetric private var blockieSize = 24
+  /// Favicon size for ERC 20 Approve transactions
+  @ScaledMetric private var faviconSize = 48
+  private let maxFaviconSize: CGFloat = 96
+  
+  private enum ViewMode: Int {
+    case transaction
+    case details
+  }
+  
+  @State private var viewMode: ViewMode = .transaction
+  
+  private var transactionType: String {
+    if confirmationStore.activeParsedTransaction.transaction.txType == .erc20Approve {
+      return Strings.Wallet.transactionTypeApprove
+    }
+    switch confirmationStore.activeParsedTransaction.transaction.txType {
+    case .erc20Approve:
+      return Strings.Wallet.transactionTypeApprove
+    case .solanaDappSignTransaction, .solanaDappSignAndSendTransaction:
+      return Strings.Wallet.solanaDappTransactionTitle
+    default:
+      return confirmationStore.activeParsedTransaction.transaction.isSwap ? Strings.Wallet.swap : Strings.Wallet.send
+    }
+  }
+  
+  /// The view for changing between available pending transactions. ex. '1 of 4 Next'
+  @ViewBuilder private var transactionsButton: some View {
+    if confirmationStore.unapprovedTxs.count > 1 {
+      let index = confirmationStore.unapprovedTxs.firstIndex(of: confirmationStore.activeParsedTransaction.transaction) ?? 0
+      HStack {
+        Text(String.localizedStringWithFormat(Strings.Wallet.transactionCount, index + 1, confirmationStore.unapprovedTxs.count))
+          .fontWeight(.semibold)
+        Button(action: confirmationStore.nextTransaction) {
+          Text(Strings.Wallet.next)
+            .fontWeight(.semibold)
+            .foregroundColor(Color(.braveBlurpleTint))
+        }
+      }
+    } else {
+      EmptyView()
+    }
```

Comment:
I think this is from before but since it's an `@ViewBuilder` we shouldn't need this else block 🙂

Reply:
fixed in [29dc327](https://github.com/brave/brave-ios/pull/6960/commits/29dc3270f2191a8677bd908927acd01b0fc1f7de)